src/systemctl/systemctl-set-environment.c

```diff
@@ -8,6 +8,58 @@
 #include "systemctl-util.h"
 #include "systemctl.h"
 
+static int json_transform_message(sd_bus_message *m, JsonVariant **ret) {
+        JsonVariant **elements = NULL;
+        size_t n_elements = 0;
+        char *text, *sep;
+        int r;
+
+        assert(m);
+        assert(ret);
+
+        r = sd_bus_message_enter_container(m, SD_BUS_TYPE_ARRAY, "s");
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        while ((r = sd_bus_message_read_basic(m, SD_BUS_TYPE_STRING, &text)) > 0) {
+                if (!GREEDY_REALLOC(elements, n_elements + 2)) {
```

Comment:
no need for a separate array, you can just create objects and merge them as you go